### PR TITLE
Update to M109, which switches to the new TypeScript 2.0 way of consu…

### PR DESCRIPTION
**Pull request by trevorsg**
…ming and providing type information.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,25 +1,18 @@
...
-  "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
-  },
+  "version": "1.109.0",
+  "description": "Visual Studio Services Client SDK. JavaScript library for Visual Studio Team Services extensions.",
   "repository": {
     "type": "git",
     "url": "https://github.com/Microsoft/vss-web-extension-sdk.git"
   },
+  "scripts": {
```
Comment (willsmythe) [inline package.json]: Would it be worthwhile to just add a "prepublish" script that builds the package prior to publishing? This pattern is suggested by the NPM docs: https://docs.npmjs.com/misc/scripts

File: package.json [hunk 2]
```diff
@@ -1,25 +1,18 @@
...
+    "visual studio team services"
   ],
-  "typescript": {
-    "definitions": [
-      "typings/rmo.d.ts",
-      "typings/tfs.d.ts",
-      "typings/vss.d.ts"
-    ]
-  },
   "author": "vsalmopen",
```
Comment (willsmythe) [inline package.json]: Author should be "Microsoft"

File: package.json (hunk 1 above)
Comment (trevorsg) [inline package.json]: Sure

File: package.json (hunk 2 above)
Comment (trevorsg) [inline package.json]: Done

File: package.json [hunk 3]
```diff
@@ -29,5 +23,17 @@
...
+  },
+  "types": "./typings/index.d.ts",
+  "dependencies": {
+    "@types/jquery": "^2.0.34",
+    "@types/jqueryui": "^1.11.31",
+    "@types/knockout": "^3.4.38",
+    "@types/q": "0.0.32",
+    "@types/react": "^0.14.54",
+    "@types/requirejs": "^2.1.28"
+  },
```
Comment (willsmythe) [inline package.json]: What about using the `main` attribute of package.json to point to VSS.SDK.js? Someone could, theoretically, pull in the VSS SDK via their own module loader, right? From NPM docs:

> The main field is a module ID that is the primary entry point to your program. That is, if your package is named foo, and a user installs it, and then does require("foo"), then your main module's exports object will be returned.
> 
> This should be a module ID relative to the root of your package folder.
> 
> For most modules, it makes the most sense to have a main script and often not much else.

Comment (trevorsg) [inline package.json]: This is a library, so I think it should be left out?

Comment (cschleiden) [inline package.json]: I think we should add it (see https://github.com/facebook/react/blob/master/packages/react/package.json, for example)

**Pull request merged**